File: back/app/schemas/Order.py
from datetime import datetime
from enum import Enum
from typing import List
from pydantic import BaseModel


class OrderStatus(str, Enum):
    PENDING = "PENDING"
    SHIPPED = "SHIPPED"
    DELIVERED = "DELIVERED"
    CANCELED = "CANCELED"


class Order(BaseModel):
    id: int
    itemNames: List[str]
    userName: str
    total: int
    orderDate: datetime
    deliveryDate: datetime
    status: OrderStatus


class OrderDataPerItem(BaseModel):
    itemId: int
    quantity: int
    total: int
    orderId: int

File: front/src/app/interfaces/Order.ts
import { Item } from "./Item";

export enum OrderStatus {
  SHIPPED = "SHIPPED",
  PENDING = "PENDING",
  DELIVERED = "DELIVERED",
  CANCELED = "CANCELED",
  ALL = "ALL"
}

export enum CartStatus {
  ADDED = "ADDED",
  DELETED = "DELETED",
  ORDERED = "ORDERED",
  PENDING = "PENDING"
}

export interface CartItem {
  id: number | null;
  status: CartStatus;
  item: Item
}

export enum FilterSortField {
  PRICE = "Price",
  ORDERDATE = "Order date",
  DELIVERYDATE = "Deliver date",
  QUANTITY = "Quantity"
}

export enum FilterSortOrder {
  ASC = "asc",
  DESC = "desc"
}

export interface Order {
  id: number;
  status: OrderStatus;
  itemNames: string[];
  total: number;
  userName: string;
  orderDate: Date;
  deliveryDate: Date;
}

export interface OrderSummary {
  itemName: string;
  basePrice: number;
  timesOrdered: number;
  totalSpend: number;
  orderDates: Date[];
}

export interface OptionType {
  value: string;
  label: string;
}
